test/e2e/apimachinery/flowcontrol.go

```diff
@@ -317,6 +335,58 @@ func createFlowSchema(f *framework.Framework, flowSchemaName string, matchingPre
 	}
 }
 
+func getFlowSchema(f *framework.Framework, name string) *flowcontrol.FlowSchema {
+	fs, err := f.ClientSet.FlowcontrolV1beta1().FlowSchemas().Get(context.TODO(), name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	return fs
+}
+
+func getPriorityLevelConfig(f *framework.Framework, name string) *flowcontrol.PriorityLevelConfiguration {
+	pl, err := f.ClientSet.FlowcontrolV1beta1().PriorityLevelConfigurations().Get(context.TODO(), name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	return pl
+}
+
+// waitForSteadyState repeatedly polls the API server to check if the newly
+// created flow schema and priority level have been seen by the APF controller
+// by checking: (1) the dangling priority level reference condition in the flow
+// schema status, and (2) metrics. The function times out after 30 seconds.
+func waitForSteadyState(f *framework.Framework, flowSchemaName string, priorityLevelName string) {
+	framework.ExpectNoError(wait.Poll(time.Second, 30*time.Second, func() (bool, error) {
+		fs := getFlowSchema(f, flowSchemaName)
+		if apihelpers.GetFlowSchemaConditionByType(fs, flowcontrol.FlowSchemaConditionDangling) != nil {
```

Comment:
There is an inversion of logic here.  The current code returns `false` if the desired condition is present.  That is backwards.  The code should return `false` if the desired condition is _not_ present.

Reply:
Done. I was certain it was the other way around, but I think I get the full picture now.